src/java/org/apache/cassandra/db/compaction/LeveledCompactionTask.java

```diff
@@ -65,7 +65,8 @@ public CompactionAwareWriter getCompactionAwareWriter(CompactionRealm realm,
     @Override
     protected boolean partialCompactionsAcceptable()
     {
-        throw new UnsupportedOperationException("This is now handled in reduceScopeForLimitedSpace");
+        // LCS allows removing L0 sstable from compaction for limited disk space. It's handled in #reduceScopeForLimitedSpace
```

Comment:
made a comment on the origin C* ticket: https://issues.apache.org/jira/browse/CASSANDRA-17272

Reply:
>  can you remind me why this is acceptable in L1 but not higher level? I remember at least that L0 is somewhat special, but is L1 too?

the `level` is compaction target level.

* if target is level 0, any sstable can be removed to reduce disk space.
* if target is level 1, any level 0 can be removed, but not sstable from level 1 to guarantee no overlapping in level 1

Reply:
> the comment here only mentions L0 (but the code includes L1 and the title of this issue does too, so I assume it's just the comment that is slightly incomplete).

Only level 0 and level 1 compaction task may contain L0 sstable.

I have updated the PR title to `removing L0 sstable from L0/L1 compaction task for limited space`

Reply:
Ok, make sense, thanks.